crates/egui_kittest/src/harness_kind.rs

```diff
@@ -0,0 +1,72 @@
+use egui::Frame;
+
+type AppKindContext<'a> = Box<dyn FnMut(&egui::Context) + 'a>;
+type AppKindUi<'a> = Box<dyn FnMut(&mut egui::Ui) + 'a>;
+
+pub enum AppKind<'a> {
+    Context(AppKindContext<'a>),
+    Ui(AppKindUi<'a>),
+}
+
+// TODO(lucasmerlin): These aren't working unfortunately :(
+// I think they should work though: https://geo-ant.github.io/blog/2021/rust-traits-and-variadic-functions/
+// pub trait IntoAppKind<'a, UiKind> {
+//     fn into_harness_kind(self) -> AppKind<'a>;
+// }
+//
+// impl<'a, F> IntoAppKind<'a, &egui::Context> for F
+// where
+//     F: FnMut(&egui::Context) + 'a,
+// {
+//     fn into_harness_kind(self) -> AppKind<'a> {
+//         AppKind::Context(Box::new(self))
+//     }
+// }
+//
+// impl<'a, F> IntoAppKind<'a, &mut egui::Ui> for F
+// where
+//     F: FnMut(&mut egui::Ui) + 'a,
+// {
+//     fn into_harness_kind(self) -> AppKind<'a> {
+//         AppKind::Ui(Box::new(self))
+//     }
+// }
+
+impl<'a> AppKind<'a> {
+    pub fn run(&mut self, ctx: &egui::Context) -> Option<egui::Response> {
+        match self {
+            AppKind::Context(f) => {
+                f(ctx);
+                None
+            }
+            AppKind::Ui(f) => Some(Self::run_ui(f, ctx, false)),
+        }
+    }
+
+    pub fn run_sizing_pass(&mut self, ctx: &egui::Context) -> Option<egui::Response> {
+        match self {
+            AppKind::Context(f) => {
+                f(ctx);
```

Comment:
Maybe we should panic here, since we're not setting any sizing-pass specialness in the `Context` case?

Reply:
I added an assert in #5313